lib/ansible/modules/windows/win_nssm.ps1

```diff
@@ -166,104 +158,86 @@ Function Nssm-Install
      if ($result.changed)
      {
         $applicationPath = (Get-Item $application).DirectoryName
-        $cmd = "nssm set ""$name"" AppDirectory ""$applicationPath"""
+        $cmd = "set ""$name"" AppDirectory ""$applicationPath"""
 
-        $results = invoke-expression $cmd
+        $nssm_result = Nssm-Invoke $cmd
 
-        if ($LastExitCode -ne 0)
+        if ($nssm_result.rc -ne 0)
         {
             $result.nssm_error_cmd = $cmd
-            $result.nssm_error_log = "$results"
+            $result.nssm_error_log = $nssm_result.stderr
             Throw "Error installing service ""$name"""
         }
      }
 }
 
-Function ParseAppParameters()
```

Comment:
We will still need to call this for backwards compatibility, would probably need a check to ensure the string is in this form so we can fire the warning.